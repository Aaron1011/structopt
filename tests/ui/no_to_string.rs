use structopt::StructOpt;

#[derive(StructOpt, Debug)]
struct Opt {
    #[structopt(short, default_value = ())]
    b: String,
}

fn main() {
    let opt = Opt::from_args();
    println!("{:?}", opt);
}
