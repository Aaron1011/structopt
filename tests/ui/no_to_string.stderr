error[E0277]: `()` doesn't implement `std::fmt::Display`
 --> $DIR/no_to_string.rs:5:40
  |
5 |     #[structopt(short, default_value = ())]
  |                                        ^^ `()` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `()`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: required because of the requirements on the impl of `std::string::ToString` for `()`
  = note: required by `std::string::ToString::to_string`
